packages/models/src/root.ts

```diff
@@ -51,7 +50,19 @@ export class Root extends Signed {
     }
   }
 
-  public equals(other: Root): boolean {
+  public addKey(key: Key, role: string): void {
+    if (!this.roles[role]) {
+      throw new ValueError(`role ${role} does not exist`);
+    }
+
+    if (!this.roles[role].keyIDs.includes(key.keyID)) {
+      this.roles[role].keyIDs.push(key.keyID);
+    }
+
+    this.keys[key.keyID] = key;
+  }
```

Comment:
New function to support the addition of new keys to the root. Needed by the new `repo-mock` package.